app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskScreen.kt

```diff
@@ -0,0 +1,134 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Done
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.example.android.architecture.blueprints.todoapp.R
+import com.example.android.architecture.blueprints.todoapp.util.getViewModelFactory
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+
+@Composable
+fun AddEditTaskScreen(
+    taskId: String?,
+    onTaskUpdate: () -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: AddEditTaskViewModel = viewModel(factory = getViewModelFactory())
+) {
+    val state = rememberAddEditTaskState(taskId, viewModel, onTaskUpdate)
+    Scaffold(
+        modifier = modifier.fillMaxSize(),
+        scaffoldState = state.scaffoldState,
+        floatingActionButton = {
+            FloatingActionButton(onClick = state::onFabClick) {
+                Icon(Icons.Filled.Done, stringResource(id = R.string.cd_save_task))
+            }
+        }
+    ) { paddingValues ->
+        val loading by viewModel.dataLoading.observeAsState(initial = false)
+        val title by viewModel.title.observeAsState(initial = "")
+        val description by viewModel.description.observeAsState(initial = "")
+
+        AddEditTaskContent(
+            loading = loading,
+            title = title,
+            description = description,
+            onTitleChanged = state::onTitleChanged,
+            onDescriptionChanged = state::onDescriptionChanged,
+            modifier = Modifier.padding(paddingValues)
+        )
+    }
+}
+
+@Composable
+private fun AddEditTaskContent(
+    loading: Boolean,
+    title: String,
+    description: String,
+    onTitleChanged: (String) -> Unit,
+    onDescriptionChanged: (String) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    if (loading) {
+        SwipeRefresh(
+            state = rememberSwipeRefreshState(loading),
```

Comment:
I know loading is always true by this point but I think it might be more clear what you are doing here if you added a comment and also just used `true` instead of `loading` in the refresh state

Reply:
Done thanks!